pyfr/solvers/baseadvecdiff/elements.py

```diff
@@ -9,35 +9,39 @@ def _scratch_bufs(self):
 
         if 'flux' in self.antialias:
             bufs |= {'scal_qpts', 'vect_qpts'}
+        else:
+            bufs |= {'grad_upts'}
 
         return bufs
 
     def set_backend(self, backend, nscalupts, nonce, linoff):
         super().set_backend(backend, nscalupts, nonce, linoff)
 
-        kernel = self._be.kernel
+        kernel, kerns = self._be.kernel, self.kernels
         kprefix = 'pyfr.solvers.baseadvecdiff.kernels'
         slicem = self._slice_mat
 
         # Register our pointwise kernels
         self._be.pointwise.register(f'{kprefix}.gradcoru')
 
+        # Set memory optimisation based on config
+        grad = self._vect_upts if 'flux' in self.antialias else self._grad_upts
```

Comment:
Check line length.

Reply:
Also provide an option to disable (it increases memory usage so we may want it off sometimes).

Reply:
I added this to `backend` as it felt more like a backend thing than a system thing. But I can move it.